hw_8/hw_8.2.js

```diff
@@ -0,0 +1,284 @@
+*/
+
+function getCompanyByCompanyId(compId) {
+	let compObj; 
+	enterprises.find(comp => comp.id === compId ? compObj = comp : false);
+	return compObj;
+}
+
+
+const addNewDepartment = (compId, departName) => {
+	const newDepart = {
+		id: getMaxId(enterprises) + 1,
+		name: departName,
+		employees_count: 0,
+	}
+	getCompanyByCompanyId(compId).departments.push(newDepart);
+}
+
+addNewDepartment(11, 'Название нового отдела');
+addNewDepartment(11, 'Название super отдела');
+console.log(enterprises[3]);
+
+
+/*
+5. Написать функцию для редактирования названия предприятия. Принимает в качестве аргумента id предприятия и новое имя предприятия.
+Пример:
+editEnterprise(1, "Новое название предприятия")
+*/
+
+const editCompanyName = (compId, newCompName) => {
+	getCompanyByCompanyId(compId).name = newCompName;
+}
+
+editCompanyName(11, 'Предприятие 4');
+console.log(enterprises[3]);
+
+
+/*
+6. Написать функцию для редактирования названия отдела. Принимает в качестве аргумента id отдела и новое имя отдела.
+Пример:
+editDepartment(7, "Новое название отдела")
+*/
+
+function getDepartByDepartId(departId) {
+	let departObj; 
+	enterprises.find(comp => comp.departments.find(depart => depart.id === departId ? departObj = depart : false));
+	return departObj;
+}
+
+
+const editDepartName = (departId, newDepartName) => {
+	getDepartByDepartId(departId).name = newDepartName;
+}
+
+editDepartName(12, 'Финансовый отдел');
+editDepartName(13, 'Коммерческий отдел');
+console.log(enterprises[3]);
+
+
+/*
+7. Написать функцию для удаления предприятия. В качестве аргумента принимает id предприятия.
+Пример:
+deleteEnterprise(1)
+*/
+
+const deleteCompanyItem = compId => {
+	const compIndex = enterprises.findIndex(comp => comp.id === compId);
+	compIndex != -1 ? enterprises.splice(compIndex, 1) : false;
+}
+
+deleteCompanyItem(5);
+console.log(enterprises);
+
+
+/*
+8. Написать функцию для удаления отдела. В качестве аргумента принимает id отдела. Удалить отдел можно только, если в нем нет сотрудников.
+Пример:
+deleteDepartment(3)
+*/
+
+const deleteDepartItem = departId => {
+	enterprises.forEach(comp => {
+		const departIndex = comp.departments.findIndex(depart => depart.id === departId && depart.employees_count == 0);
+		departIndex != -1 ? comp.departments.splice(departIndex, 1) : false;
+	})
+}
+
+deleteDepartItem(12);
+console.log(enterprises[2]);
+
+
+/*
+9. Написать функцию для переноса сотрудников между отделами одного предприятия. В качестве аргумента принимает два значения: id отдела, из которого будут переноситься сотрудники и id отдела, в который будут переноситься сотрудники).
+Пример:
+moveEmployees(2, 3)
+*/
+
+const moveEmployees = (startDepartId, endDepartId) => {
+	let startDepart;
+	let endDepart;
+	enterprises.forEach(comp => {
+		comp.departments.forEach(depart => {
+			if (depart.id === startDepartId) {
+				startDepart = depart;
+			} else if (depart.id === endDepartId) {
+				endDepart = depart;
+			} 
+		})
```

Comment:
```suggestion
	const startDepart = getDepartmentById(startDepartId)
	const endDepart = getDepartmentById(endDepartId)
```
Use your own helpers! And don't forget that your helper is named kinda like getDepartByDepartId